ui/tests/integration/components/auth-jwt-test.js

```diff
@@ -127,35 +122,46 @@ module('Integration | Component | auth jwt', function (hooks) {
   });
 
   test('oidc: test role: it renders', async function (assert) {
+    // setting the path also fires off a request to auth_url but this happens inconsistently in tests
+    // setting here so it doesn't affect the postCount because it's not relevant to what's being tested
+    this.set('selectedAuthPath', 'foo');
     let postCount = 0;
     this.server.post('/auth/:path/oidc/auth_url', (_, request) => {
       postCount++;
       const { role } = JSON.parse(request.requestBody);
-      if (['test', 'okta', 'bar'].includes(role)) {
-        const auth_url = role === 'test' ? 'http://example.com' : role === 'okta' ? 'http://okta.com' : '';
-        return {
-          data: { auth_url },
-        };
-      }
-      const errors = role === 'foo' ? ['role "foo" could not be found'] : [ERROR_JWT_LOGIN];
-      return overrideResponse(400, { errors });
+      const auth_url = role === 'test' ? 'http://example.com' : role === 'okta' ? 'http://okta.com' : '';
+      return {
+        data: { auth_url },
+      };
     });
     await renderIt(this);
     await settled();
+    await fillIn(AUTH_FORM.roleInput, 'test');
+    assert
+      .dom(AUTH_FORM.input('jwt'))
+      .doesNotExist('does not show jwt token input if role matches OIDC login url');
+    assert.dom(AUTH_FORM.login).hasText('Sign in with OIDC Provider');
+    await fillIn(AUTH_FORM.roleInput, 'okta');
+    // 1 for initial render, 1 for each time role changed = 3
+    assert.strictEqual(postCount, 3, 'fetches the auth_url when the role changes');
+    assert.dom(AUTH_FORM.login).hasText('Sign in with Okta', 'recognizes auth methods with certain urls');
+  });
+
+  test('oidc: it fetches auth_url when path changes', async function (assert) {
+    assert.expect(2);
+    this.set('selectedAuthPath', 'foo');
+    await renderIt(this);
+    // auth_url is requested on initial render so stubbing after rendering the component
+    // to test auth_url is called when the :path changes
+    this.server.post('/auth/:path/oidc/auth_url', (_, request) => {
+      assert.true(true, 'request is made to auth_url');
+      assert.strictEqual(request?.params?.path, 'foo', 'request params are { path: foo }');
+      return {
+        data: { auth_url: '' },
+      };
+    });
     this.set('selectedAuthPath', 'foo');
-    await component.role('test');
```

Comment:
I attempted replacing all `component` uses but that introduced other flakiness that felt out of scope for this whack-a-mole test flakiness